src/main/java/org/dependencytrack/tasks/BomUploadProcessingTask.java

```diff
@@ -161,8 +161,11 @@ public void inform(final Event e) {
                     // vulnerability analysis completed.
                     vae.onSuccess(new NewVulnerableDependencyAnalysisEvent(newComponents));
                 }
+                // Wait for RepositoryMetaEvent after VulnerabilityAnalysisEvent,
+                // as both might be needed in policy evaluation
```

Comment:
I realize this is required for some policy conditions, however the downside is that CI pipelines will spend **a lot** more time waiting for results. 

In contrast to vulnerability analysis, repo meta analysis cannot make use of batch APIs. The repo meta analysis may also involve requesting data from multiple repositories for the same component. The increase in latency may not be noticeable for small-ish projects, but it definitely will be for larger ones. Also in scenarios where many BOMs are uploaded concurrently. My concern with this change is that it will cause lots of timeouts in our user's pipelines.

The part we're currently missing to make this a viable option is that we can't report to callers of `/api/v1/bom/token/<TOKEN>` *what* processing steps are done, and which are still pending. So users can't opt out of waiting for repo meta analysis to complete, if all they care about is the vulnerabilities.

For now, what we could do is execute policy evaluation after both the `RepositoryMetaEvent` *and* `VulnerabilityAnalysisEvent`. The `RepositoryMetaEvent` would not be attached to `VulnerabilityAnalysisEvent`, so there wouldn't be an impact on CI pipelines. Granted, there's also a non-zero risk of race conditions due to the duplicate `PolicyEvaluationEvent`.